In [1]:
print(2)

2


In [5]:
import gymnasium

from huggingface_sb3 import load_from_hub, package_to_hub
from huggingface_hub import (
    notebook_login,
) 

from stable_baselines3 import PPO
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.evaluation import evaluate_policy
from stable_baselines3.common.monitor import Monitor

/home/sam/Github/365DaysOfData/17-Deep-Reinfrocement-Learning/rlenv/lib/python3.13/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [21]:
import gymnasium as gym

In [22]:
env = gym.make("LunarLander-v3", render_mode="human")


# now we use the PPO model, combination of value based rl and policy based rl
model = PPO('MlpPolicy', env, verbose=1)


model.learn(total_timesteps=10000)

Using cuda device
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.


/home/sam/Github/365DaysOfData/17-Deep-Reinfrocement-Learning/rlenv/lib/python3.13/site-packages/stable_baselines3/common/on_policy_algorithm.py:150: UserWarning: You are trying to run PPO on the GPU, but it is primarily intended to run on the CPU when not using a CNN policy (you are using ActorCriticPolicy which should be a MlpPolicy). See https://github.com/DLR-RM/stable-baselines3/issues/1245 for more info. You can pass `device='cpu'` or `export CUDA_VISIBLE_DEVICES=` to force using the CPU.Note: The model will train, but the GPU utilization will be poor and the training might take longer than on CPU.
  warnings.warn(


---------------------------------
| rollout/           |          |
|    ep_len_mean     | 95.4     |
|    ep_rew_mean     | -180     |
| time/              |          |
|    fps             | 47       |
|    iterations      | 1        |
|    time_elapsed    | 42       |
|    total_timesteps | 2048     |
---------------------------------
------------------------------------------
| rollout/                |              |
|    ep_len_mean          | 95.1         |
|    ep_rew_mean          | -163         |
| time/                   |              |
|    fps                  | 47           |
|    iterations           | 2            |
|    time_elapsed         | 86           |
|    total_timesteps      | 4096         |
| train/                  |              |
|    approx_kl            | 0.0061743045 |
|    clip_fraction        | 0.0145       |
|    clip_range           | 0.2          |
|    entropy_loss         | -1.38        |
|    explained_variance   | -0.00398     |
|    learning_r

In [23]:
# We create our environment with gym.make("<name_of_the_environment>")
env = gym.make("LunarLander-v3", render_mode="human")
env.reset()
print("_____OBSERVATION SPACE_____ \n")
print("Observation Space Shape", env.observation_space.shape)
print("Sample observation", env.observation_space.sample())  # Get a random observation

print("\n _____ACTION SPACE_____ \n")
print("Action Space Shape", env.action_space.n)
print("Action Space Sample", env.action_space.sample())  # Take a random action

env.close()

_____OBSERVATION SPACE_____ 

Observation Space Shape (8,)
Sample observation [-1.2647774   1.2805381   1.9561454  -8.315251   -4.395654    2.9619045
  0.34899753  0.18285781]

 _____ACTION SPACE_____ 

Action Space Shape 4
Action Space Sample 3


In [29]:
env = gym.make("LunarLander-v3" ,render_mode = "rgb_array")
model = PPO(
    policy="MlpPolicy",
    env=env,
    n_steps=1024,
    batch_size=64,
    n_epochs=4,
    gamma=0.999,
    gae_lambda=0.98,
    ent_coef=0.01,
    verbose=1,
)

Using cuda device
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.


/home/sam/Github/365DaysOfData/17-Deep-Reinfrocement-Learning/rlenv/lib/python3.13/site-packages/stable_baselines3/common/on_policy_algorithm.py:150: UserWarning: You are trying to run PPO on the GPU, but it is primarily intended to run on the CPU when not using a CNN policy (you are using ActorCriticPolicy which should be a MlpPolicy). See https://github.com/DLR-RM/stable-baselines3/issues/1245 for more info. You can pass `device='cpu'` or `export CUDA_VISIBLE_DEVICES=` to force using the CPU.Note: The model will train, but the GPU utilization will be poor and the training might take longer than on CPU.
  warnings.warn(


In [30]:
model.learn(total_timesteps=1000000)
model_name = "ppo-LunarLander-sam"
model.save(model_name)

---------------------------------
| rollout/           |          |
|    ep_len_mean     | 86       |
|    ep_rew_mean     | -135     |
| time/              |          |
|    fps             | 1055     |
|    iterations      | 1        |
|    time_elapsed    | 0        |
|    total_timesteps | 1024     |
---------------------------------
-------------------------------------------
| rollout/                |               |
|    ep_len_mean          | 93.8          |
|    ep_rew_mean          | -134          |
| time/                   |               |
|    fps                  | 1044          |
|    iterations           | 2             |
|    time_elapsed         | 1             |
|    total_timesteps      | 2048          |
| train/                  |               |
|    approx_kl            | 0.00015830656 |
|    clip_fraction        | 0             |
|    clip_range           | 0.2           |
|    entropy_loss         | -1.39         |
|    explained_variance   | -0.00243      |


In [31]:
eval_env = Monitor(gym.make("LunarLander-v3"))
mean_reward, std_reward = evaluate_policy(model, eval_env, n_eval_episodes=10, deterministic=True)
print(f"mean_reward={mean_reward:.2f} +/- {std_reward}")

mean_reward=226.56 +/- 78.56714748934611


In my case, I got a mean_reward=226.56 +/- 78.56714748934611 after training for 1 million steps for 30 mins, which means that our lunar lander agent is ready to land on the moon


In [ ]:
notebook_login()
!git config --global credential.helper store

In [ ]:
import gymnasium as gym
from stable_baselines3.common.vec_env import DummyVecEnv
from stable_baselines3.common.env_util import make_vec_env

from huggingface_sb3 import package_to_hub

repo_id = "paudelsamir/ppo-LunarLander-sam"

env_id = "LunarLander-v3"

eval_env = DummyVecEnv([lambda: gym.make(env_id, render_mode="rgb_array")])

model_architecture = "PPO"

commit_message = "Initial commit with trained PPO model on LunarLander-v3"

package_to_hub(model=model, 
               model_name=model_name,
               model_architecture=model_architecture, 
               env_id=env_id, 
               eval_env=eval_env,
               repo_id=repo_id,
               commit_message=commit_message)